### ISSUE TITLE
Trouble building with macports

### ISSUE THREAD

**Issue by dubiousjim** [rust-lang/git2-rs#563]
I'm trying to build a package (cargo-outdated) that depends on this package. My build platform is OS X (an older one, 10.10.5), using macports. The build fails, as [reported here](https://github.com/kbknapp/cargo-outdated/issues/187). Given the error messages, I thought the underlying issue might be with this package rather than with cargo-outdated?

**Comment by dubiousjim** [comment]
Issue #263 might be related.

**Comment by dubiousjim** [comment]
One of the later comments on issue #263 suggested building with `RUSTFLAGS="-L/usr/lib/libiconv"`, to avoid linking against the macports version of libiconv. I'm using a version of OS X where the libraries are just in /usr/lib, not in /usr/lib/libiconv. So I instead tried `port deactivate -f libiconv`, then `cargo build --release` (for cargo-outdated, planning to reactivate `libiconv` afterwards). But that starts a chain of other linked macports libraries complaining that they depend on newer versions than are present in `/usr/lib` (and when they're deactivated too, other libraries complain...).

So I tried `RUSTFLAGS="-L/usr/lib" cargo build --release` and when that didn't work, `RUSTFLAGS="-L/usr/lib -L/opt/local/lib" cargo build --release`. But these both failed, with lots of linking errors. So what I finally tried was to create a local directory `./lib`, link to the system `libiconv.*` libraries from there, and do `RUSTFLAGS="-L./lib" cargo build --release` (for cargo-outdated, the package that was including various pieces of this package). That built successfully.
